frontend/views/components/Contribution.vue

```diff
@@ -118,32 +134,35 @@ export default {
       this.isAdding = false
       this.isEditing = false
       this.isFilled = false
-      this.hasError = false
     },
     handleEnter (e) {
       if (this.isAdding && !this.isFilled) {
         this.setError()
         return false
       }
-
       return this.isFilled ? this.handleSubmit() : this.handleDelete()
     },
     handleDelete () {
-      this.$emit('new-value', this.$refs.input.value)
+      this.$emit('new-value', 'nonMonetaryRemove', this.initialValue)
       this.cancel()
     },
     handleSubmit () {
-      const text = this.$refs.input.value
-
-      if (text.trim() === '') {
-        this.setError()
-      } else {
-        this.$emit('new-value', text)
+      if (this.$v.form.$invalid) {
         this.cancel()
+      } else {
+        if (this.isAdding) {
+          this.$emit('new-value', 'nonMonetaryAdd', this.form.contribution)
+          this.isAdding = false
+        }
+        if (this.isEditing) {
+          this.$emit('new-value', 'nonMonetaryEdit', {
+            replace: this.initialValue || false,
```

Comment:
`replace` must always have a value that exists in the list, otherwise there will be an exception thrown (so `false` cannot be here, and `this.initialValue` must have been previously added via `nonMonetaryAdd`)